x-pack/plugins/viz_editor/server/query/select_operations.ts

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+/*********************************************************************************************
+ * Logic to convert our query shape into an Elastic search query. It takes a flat list
+ * of selection criteria, and converts them into the various-- sometimes heirardchical--
+ * shapes expected by the Elasticsearch DSL.
+ *********************************************************************************************/
+
+import {
+  AvgOperation,
+  chunkBy,
+  ChunkResult,
+  Column,
+  CountOperation,
+  DateHistogramOperation,
+  isEmpty,
+  partition,
+  Query,
+  SelectOperation,
+} from '../../common';
+
+/**
+ * A handful of transformation functions for a select operation,
+ * used to build up an ES query and flatten the results.
+ */
+export interface SelectDefinition {
+  toEsAgg?: (op: any) => any;
```

Comment:
I believe this should also have a `toEsDocs` function, right?

Reply:
Maybe? I'm not sure what such a function would do. If we end up needing it, we can add it then.